File: InterconnectBackend/Models/Database/BootableDiskModel.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Models.Database
{
    /// <summary>
    /// Database model representing a bootable disk.
    /// </summary>
    public class BootableDiskModel
    {
        /// <summary>
        /// Disk identifier.
        /// </summary>
        public required int Id { get; set; }
        
        /// <summary>
        /// Disk name.
        /// </summary>
        [Column(TypeName = "varchar(255)")]
        public required string Name { get; set; }
        
        /// <summary>
        /// Operating system version on the disk.
        /// </summary>
        [Column(TypeName = "varchar(32)")]
        public string? Version { get; set; }
        
        /// <summary>
        /// Path to the disk file.
        /// </summary>
        [Column(TypeName = "varchar(255)")]
        public string? Path { get; set; }
    }
}

File: InterconnectFrontend/src/models/BootableDiskModel.ts
/**
 * Represents a bootable disk available in the system.
 */
interface BootableDiskModel {
    /** Unique identifier for the bootable disk */
    id: number;
    /** Name of the bootable disk */
    name: string;
    /** Version of the bootable disk */
    version: string;
}

export type { BootableDiskModel };
